src/contexts/interactiveComponentsContext.tsx

```diff
@@ -6,32 +6,55 @@ export interface InteractiveComponentsState {
   toggleSidebar: () => void;
   selectedIndex: string;
   selectIndex: (index: string) => void;
+  isProfileModalOpened: boolean;
+  closeProfileModal: () => void;
+  openProfileModal: () => void;
 }
 
 const InteractiveComponentsDefaultState: InteractiveComponentsState = {
   isSidebarOpened: true,
   toggleSidebar: () => null,
   selectedIndex: '',
   selectIndex: () => null,
+  isProfileModalOpened: false,
+  closeProfileModal: () => null,
+  openProfileModal: () => null,
```

Comment:
The profile modal is way too specific to have its api part of the InteractiveComponentsContext.
The modal controls that would be part of the InteractiveComponentsContext are those of a generic modal that we reuse across application. This modal is way too specific and will probably never be opened from more than one place. 
Move the logic to open/close the ProfileModal to the specific part of the ui from where it is opened

Reply:
Have updated the logic in the **interactiveComponentsContext**, only for the **modal** for now. If that is good to go, will create a separate ticket to address the changes for the whole **interactiveComponentsContext**.